### ISSUE TITLE
Practical examples?

### ISSUE THREAD

**Issue by RamblingCookieMonster** [Cidney/Cidney#2]
Hi!

A few practical examples in the docs or the wiki might be a helpful way to illustrate how to use this.  For example:
- A simple example showing how to test a PowerShell tool with Pester and ScriptAnalyzer, and deploy it with a script or a tool (shameless [plug](http://ramblingcookiemonster.github.io/PSDeploy/))
- An example showing how to test some flavor of a .NET project, with build, test, and deploy as part of the mix.

No need for anything fancy, just barebones examples that folks could use to get started, and get an idea of how this might be used in the wild.  If you can, a [gif](http://blog.bahraniapps.com/gifcam/) illustrating the outcome of each can often be valuable for end users; take a look at @dfinke's [ImportExcel](https://github.com/dfinke/ImportExcel) for a few examples.

I know these might sound a little silly and tangential to the code itself, but this might get more eyes on the code and solution, giving you more feedback and ideas.

Cheers!

**Comment by mattmcnabb** [comment]
I'll tack on to that: how would you expect to execute these pipelines? For instance, most CI products I'm aware of include some sort of a task runner that orchestrates the tasks. With Cidney, would you use  Task Scheduler, web hooks, etc., or would you just run the pipeline manually?

**Comment by RobertKozak** [comment]
I agree and I am working on it. I started another project called Cidney.Commands where I put code to get source from TFS and Nuget package restore. This week I should be able to msbuild and that should help with some good real world examples.

I still have more documentation and samples coming.

**Comment by RobertKozak** [comment]
@mattmcnabb, That is a great question. Today I checked in code that changed Pipeline: so that it does not run right away. Prior to this check in it just started with what was inside the scriptblock and ran it. Now it creates a function which can run with Invoke-Cidney. Also added Get-CidneyPipeline and Remove-CidneyPipeline. 

I also have a When: Keyword that uses eventing so it can respond to Events. I didnt put a lot of effort in this yet and I want to go back and play with this some more.

I also am thinking of adding the ability to respond to webhooks. I have some ideas on this but I probably need a couple of weeks to flesh it out.   

So long story short prior to today you had to just run it manually. Now there are more options.

**Comment by RobertKozak** [comment]
And thanks for the suggestions. This project is less than 2 weeks old and I wrote this because I need it. I have a vision and I really hope it can do what I want. I love the feedback because at the moment I am working in a vacuum.

**Comment by mattmcnabb** [comment]
Great answer! So in other words, we may be able to leverage Cidney from a variety of orchestration platforms? I'm thinking Azure Automation at the moment, but I'll have to see how I would achieve that. Thanks!

**Comment by RobertKozak** [comment]
I have invited you both as members to Cidney if you want to join.

**Comment by RobertKozak** [comment]
If you can run powershell you can run a Cidney Pipeline.
